### ISSUE TITLE
Should this project be a part of Apache Impala?

### ISSUE THREAD

**Issue by wesm** [cloudera/impyla#297]
I've noticed that Impyla is languishing. I never really understood the split between the Python-based Impala shell and impyla. Perhaps it would be a good time to shift efforts to deliver an enterprise-grade Python client to the Apache project?

**Comment by gregrahn** [comment]
I would agree with this vision and would like to see impala-shell use it as well which would be a forcing function to maintain it as a first class citizen.  Perhaps even use [hs2client](https://github.com/cloudera/hs2client) underneath impyla.  There is also a fair amount of work required for impala-shell to move to Python 3 and off Beeswax and move to HS2 (via impyla) as this also impacts impala test infra AFAIK.

**Comment by timarmstrong** [comment]
The one advantage to keeping this separate is that the barrier for contributions is much lower (compared to making a contribution to mainline Impala). But if we had a feature-complete enterprise-grade client we'd be less reliant on community contributions anyway.

We do run a bunch of Impala tests via this client, particularly since https://issues.apache.org/jira/browse/IMPALA-7647. One thing that's a little awkward about running Impala tests through a DBAPI client is that timestamps get rounded to microseconds. E.g. see the difference in results here between the tests run through Impyla and the tests run directly via thrift: https://gerrit.cloudera.org/#/c/11546/9/testdata/workloads/functional-query/queries/QueryTest/analytic-fns.test
